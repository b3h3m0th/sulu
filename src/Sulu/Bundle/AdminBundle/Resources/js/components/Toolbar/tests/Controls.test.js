/* eslint-disable flowtype/require-valid-file-annotation */
import {render} from '@testing-library/react';
import React from 'react';
import Controls from '../Controls';
import Button from '../Button';

const clickSpy = jest.fn();

test('Render controls', () => {
    const {container} = render(<Controls />);
    expect(container).toMatchSnapshot();
});

test('Render controls with children', () => {
    const {container} = render(<Controls><Button onClick={clickSpy}>Test</Button></Controls>);
    expect(container).toMatchSnapshot();
});

test('Render growing controls', () => {
    const {container} = render(<Controls grow={true}></Controls>);
    expect(container).toMatchSnapshot();
});
