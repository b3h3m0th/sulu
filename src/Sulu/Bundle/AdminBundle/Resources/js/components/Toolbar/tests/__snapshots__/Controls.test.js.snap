// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render controls 1`] = `
<div>
  <div
    class="controls light"
  />
</div>
`;

exports[`Render controls with children 1`] = `
<div>
  <div
    class="controls light"
  >
    <button
      class="button light"
      type="button"
    >
      <span
        class="label"
      >
        Test
      </span>
    </button>
  </div>
</div>
`;

exports[`Render growing controls 1`] = `
<div>
  <div
    class="controls light grow"
  />
</div>
`;
